tests/client/test_query.py

```diff
@@ -61,3 +61,139 @@ def test_get_container_sources():
         'upstream_urls': [yum_utils_upstream_url, etcd_upstream_url]
     }
     assert query.get_container_content_sources(container_build_id) == expected
+
+
+def test_get_current_and_previous_versions():
+    """Test the get_current_and_previous_versions function."""
+    go = Component(
+        canonical_name='golang', canonical_type='generic', canonical_namespace='redhat').save()
+    next_sl = None
+    url = 'git://pkgs.domain.local/rpms/golang?#fed96461b05c0078e537c93a3fe974e8b334{version}'
```

Comment:
May In know why there is not any separator between hash and `{version}`?

Reply:
It was just a way to create an identifiable unique commit hash per source URL. Nothing more than that.